spock/utils.py

```diff
@@ -45,6 +46,188 @@ def add_vector(self, x=None, y=None, z=None, vec=None):
 	def __str__(self):
 		return "({:.2f}, {:.2f}, {:.2f})".format(self.x, self.y, self.z)
 
+	def to_Vector3(self):
+		"""
+		Return a Vector3 object from this one
+		"""
+		return Vector3(self.x, self.y, self.z)
+
+
+class BaseVector(object):
+	_internal_vec_type = list
+	def __init__(self, *values):
+		self.vector = self._internal_vec_type(values)
+
+	def __iter__(self):
+		return self.vector.__iter__()
+
+	def __repr__(self):
+		return "%s(" % self.__class__.__name__ + ", ".join(map(str, self.vector)) + ")"
+
+	__str__ = __repr__
+
+	def __setitem__(self, key, value):
+		self.vector[key] = value
+
+	def __getitem__(self, item):
+		return self.vector[item]
+
+	def __len__(self):
+		return len(self.vector)
+
+
+class CartesianVector(BaseVector):
+
+	# Math operations
+	# Is __abs__ really useful ?
```

Comment:
When I wrote this code, I make it super generic, but it is your project and I will trust you.